### DISCUSSION TITLE
Example of email_passwordless login provider

### DISCUSSION THREAD

**Discussion by serejke** [Web3Auth/Web3Auth#719] [Help]
Hello, Web3Auth community!

I'm implementing an integration of Web3Auth into our DApp.
I've managed to integrate with custom Google auth without issues.

Now I want to implement the email sign up.
I've seen there is a `email_passwordless` login provider of the `OpenLogin` adapter.
But I cannot find any example of that provider, and what needs to be configured on the backend side and in the dashboard.

My goal is to have:
1) Email input on the /login page.
2) User enters an email and submits the form.
3) User received confirmation email.
4) User goes to the mail box and clicks the confirmation button.
5) User gets logged in with Web3Auth.

**Comment by serejke** [comment]
For what I see now, I can implement the Email verification:

1) Deploy a custom JWT verifier and provide custom JWKs on my app
2) Implement the email verification flow myself: send email with a temporary verification link which points to my app.
3) Once the user clicks the link, the app issues JWT token
4) Pass that JWT token into `web3AuthCore.connectTo` with `loginProvider = jwt`?

But I hope there is an out-of-the-box method from Web3Auth. In the [demo](https://demo-app.web3auth.io/) it works as what I need, but I cannot understand how...

**Comment by rinzler99** [comment]
Hey, 
1. Go to [demo](https://demo-app.web3auth.io/) app [source](https://github.com/Web3Auth/Web3Auth/tree/master/demo/vue-app) code.
2. Check how openlogin adapter is configured with custom verifiers.
3. Refer the [doc](https://web3auth.io/docs/sdk/web/openlogin#web3authopenlogin-adapter) for setting up openlogin adapter with desired login configuration in your case it is only email_passwordless.

**Comment by serejke** [comment]
@rinzler99 hey, I've just looked through all the demo code and I couldn't find any configuration for the email verifier. 
Docs you provided barely mention email verification.

Could you point to the exact code/implementation or docs? Or maybe explain how the email flow works in a nutshell...
I appreciate your help a lot!

**Reply by shahbaz17** [reply]
Hey @serejke 

Please take a look at https://web3auth.io/docs/guides/auth0#enable-email-passwordless.

This is using Auth0 as an email login provider.  And this is how we set up the Verifier https://web3auth.io/docs/guides/auth0#create-email-passwordless-verifier

And for custom JWT, use https://web3auth.io/docs/custom-authentication/byo-jwt-providers#using-jose to set up the Verifier.
